SafeguardSessions/test/Unit/schema/TestFlatten.query.pq

```diff
@@ -4,8 +4,120 @@ SchemaUtils = Extension.ImportModule("SchemaUtils.pqm");
 
 SchemaUtils.FieldNamesOfType = SchemaUtils[FieldNamesOfType];
 SchemaUtils.IsFlat = SchemaUtils[IsFlat];
+SchemaUtils.FlattenLists = SchemaUtils[FlattenLists];
+SchemaUtils.FlattenRecords = SchemaUtils[FlattenRecords];
 SchemaUtils.Flatten = SchemaUtils[Flatten];
 
+TestFlatteningLists = () =>
+    let
+        AssertResult = (description as text, t as table, fieldNames as list, expectedTable as table) =>
+            Fact(description, expectedTable, SchemaUtils.FlattenLists(t, fieldNames)),
+        fieldNames = {"list_field"},
+        cases = {
+            {
+                "List field is flattened correctly",
+                #table({"list_field"}, {{{"1", "2"}}, {{"3", "4"}}}),
+                fieldNames,
+                #table(type table [list_field = any], {{"1,2"}, {"3,4"}})
+            },
+            {
+                "List field with empty list is flattened correctly",
+                #table({"list_field"}, {{{}}, {{"3", "4"}}}),
+                fieldNames,
+                #table(type table [list_field = any], {{""}, {"3,4"}})
+            },
+            {
+                "List field with null value is flattened correctly",
+                #table({"list_field"}, {{{"1", "2"}}, {null}}),
+                fieldNames,
+                #table(type table [list_field = any], {{"1,2"}, {null}})
+            },
+            {
+                "Multiple list fields are flattened correctly",
+                #table({"list_field_1", "list_field_2"}, {{{"1", "2"}, {"4", "3"}}, {{"3", "4"}, {"2", "1"}}}),
+                {"list_field_1", "list_field_2"},
+                #table(type table [list_field_1 = any, list_field_2 = any], {{"1,2", "4,3"}, {"3,4", "2,1"}})
+            }
+        },
+        facts = ProvideDataForTest(cases, AssertResult)
+    in
+        facts;
+
+TestFlattenListsThrowsError = () =>

```

Comment:
Sorry, but again I have problems with the naming. I think what we test here is that converting lists with non-text values are handled. Can you name it more appropriately?

Reply:
Done!